modules/csvparser/filterx-func-format-csv.c

```diff
@@ -0,0 +1,255 @@
+          success = filterx_dict_iter(csv_data, _append_to_buffer, user_data);
+        }
+    }
+  else
+    {
+      filterx_eval_push_error("input must be a dict or list. " FILTERX_FUNC_FORMAT_CSV_USAGE, s, csv_data);
+      filterx_object_unref(csv_data);
+      return NULL;
+    }
+
+  filterx_object_unref(csv_data);
+  return success ? filterx_string_new(formatted->str, formatted->len) : NULL;
+}
+
+static void
+_free(FilterXExpr *s)
+{
+  FilterXFunctionFormatCSV *self = (FilterXFunctionFormatCSV *) s;
+
+  filterx_expr_unref(self->input);
+  filterx_expr_unref(self->columns);
+  filterx_function_free_method(&self->super);
+}
+
+static FilterXExpr *
+_extract_columns_expr(FilterXFunctionArgs *args, GError **error)
+{
+  return filterx_function_args_get_named_expr(args, FILTERX_FUNC_FORMAT_CSV_ARG_NAME_COLUMNS);
+}
+
+static gboolean
+_extract_delimiter_arg(FilterXFunctionFormatCSV *self, FilterXFunctionArgs *args, GError **error)
+{
+  gboolean exists;
+  gsize delimiter_len;
+  const gchar *delimiter = filterx_function_args_get_named_literal_string(args,
+                           FILTERX_FUNC_FORMAT_CSV_ARG_NAME_DELIMITER,
+                           &delimiter_len, &exists);
+  if (!exists)
+    return TRUE;
+
+  if (!delimiter)
+    {
+      g_set_error(error, FILTERX_FUNCTION_ERROR, FILTERX_FUNCTION_ERROR_CTOR_FAIL,
+                  "delimiter must be a string literal. " FILTERX_FUNC_FORMAT_CSV_USAGE);
+      return FALSE;
+    }
+
+  if (delimiter_len != 1)
+    {
+      g_set_error(error, FILTERX_FUNCTION_ERROR, FILTERX_FUNCTION_ERROR_CTOR_FAIL,
+                  "delimiter must be a single character. " FILTERX_FUNC_FORMAT_CSV_USAGE);
+      return FALSE;
+    }
+
+  self->delimiter = delimiter[0];
+  return TRUE;
+}
+
+static gboolean
+_extract_arguments(FilterXFunctionFormatCSV *self, FilterXFunctionArgs *args, GError **error)
+{
+  gsize args_len = filterx_function_args_len(args);
+  if (args_len != 1)
+    {
+      g_set_error(error, FILTERX_FUNCTION_ERROR, FILTERX_FUNCTION_ERROR_CTOR_FAIL,
+                  "invalid number of arguments. " FILTERX_FUNC_FORMAT_CSV_USAGE);
+      return FALSE;
+    }
+
+  self->input = filterx_function_args_get_expr(args, 0);
+  if (!self->input)
+    {
+      g_set_error(error, FILTERX_FUNCTION_ERROR, FILTERX_FUNCTION_ERROR_CTOR_FAIL,
+                  "input must be set. " FILTERX_FUNC_FORMAT_CSV_USAGE);
+      return FALSE;
+    }
+
+  if (!_extract_delimiter_arg(self, args, error))
+    return FALSE;
+
+  self->columns = _extract_columns_expr(args, error);
+
+  return TRUE;
+}
+
+FilterXFunction *
+filterx_function_format_csv_new(const gchar *function_name, FilterXFunctionArgs *args, GError **error)
+{
+  FilterXFunctionFormatCSV *self = g_new0(FilterXFunctionFormatCSV, 1);
+  filterx_function_init_instance(&self->super, function_name);
+
+  self->super.super.eval = _eval;
+  self->super.super.free_fn = _free;
+  self->delimiter = ',';
```

Comment:
the default delimiter for the parse_csv() function is a space, so I think we should match that here.